### PR TITLE
Zihan/wd 67 ionic user location

### PR DESCRIPTION
**Pull request by zihan0**

### CODE REVIEW

File: UI/basicGUI/src/pages/home/home.ts [hunk 1]
```diff
@@ -13,19 +13,52 @@ export class HomePage {
...
     this.loadmap();
   }
 
+   getLocation(){
+     this.map.locate({
+     }).once('locationfound', (e) => {
+      this.currentLatlng = e.latlng
+
+      let markerGroup = leaflet.featureGroup();
+      if (this.marker) { // check
```
Comment (GrennanM) [inline UI/basicGUI/src/pages/home/home.ts]: should you remove comments on line 32 and 33?
"// check"

**Review by GrennanM — COMMENTED** [review]
Change comments

File: UI/basicGUI/src/pages/home/home.ts [hunk 2]
```diff
@@ -13,19 +13,52 @@ export class HomePage {
...
+      })
+  }
+
+    showLocation(){
+      this.getLocation()
+
+      let alertOfLoc = this.alertCtrl.create({
+        title:"YOUR CURRENT LOCATION",
+        subTitle: "Latitude:" + this.currentLatlng.lat+" Longitude: " +this.currentLatlng.lng,
+        buttons:['GOT IT']
```
Comment (bonellon) [inline UI/basicGUI/src/pages/home/home.ts:50]: very beautiful implementation, congrats

**Review by bonellon — DISMISSED** [review]

File: UI/basicGUI/src/pages/home/home.ts (hunk 1 above)
Comment (zihan0) [reply UI/basicGUI/src/pages/home/home.ts]: i explained the code there with a new comment, let me know if i should delete it

**Review by GrennanM — APPROVED** [review]

File: UI/basicGUI/src/pages/home/home.ts [hunk 3]
```diff
@@ -27,10 +27,10 @@ export class HomePage {
      this.map.locate({
      }).once('locationfound', (e) => {
       this.currentLatlng = e.latlng
-
+      //remove the marker before adding a new one
```
Comment (bonellon) [inline UI/basicGUI/src/pages/home/home.ts:30]: why?

**Review by bonellon — APPROVED** [review]

**Pull request merged**